crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -329,72 +329,100 @@ impl Compilation {
     }
   }
 
-  pub fn file_dependencies(&self) -> impl Iterator<Item = &PathBuf> {
-    self
+  pub fn file_dependencies(
+    &self,
+  ) -> (
+    impl Iterator<Item = &PathBuf>,
+    impl Iterator<Item = &PathBuf>,
+    impl Iterator<Item = &PathBuf>,
+  ) {
+    let all_files = self
       .make_artifact
       .file_dependencies
       .files()
-      .chain(
```

Comment:
file_dependencies in module_executor has been added to make_artifact by generate module when import_module executed.